test/test_transforms_tensor.py

```diff
@@ -710,6 +568,121 @@ def test_autoaugment(self):
+    transformed_p_list = fn(pil_img, **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_p_list)
+    assert len(transformed_t_list) == out_length
+    for transformed_tensor, transformed_pil_img in zip(transformed_t_list, transformed_p_list):
+        _assert_equal_tensor_to_pil(transformed_tensor, transformed_pil_img)
+
+    transformed_t_list_script = scripted_fn(tensor.detach().clone(), **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_t_list_script)
+    assert len(transformed_t_list_script) == out_length
+    for transformed_tensor, transformed_tensor_script in zip(transformed_t_list, transformed_t_list_script):
+        assert_equal(transformed_tensor, transformed_tensor_script)
+
+    # test for class interface
+    fn = getattr(T, method)(**meth_kwargs)
+    scripted_fn = torch.jit.script(fn)
+    output = scripted_fn(tensor)
+    assert len(output) == len(transformed_t_list_script)
+
+    # test on batch of tensors
+    batch_tensors = _create_data_batch(height=23, width=34, channels=3, num_samples=4, device=device)
+    torch.manual_seed(12)
+    transformed_batch_list = fn(batch_tensors)
+
+    for i in range(len(batch_tensors)):
+        img_tensor = batch_tensors[i, ...]
+        torch.manual_seed(12)
+        transformed_img_list = fn(img_tensor)
+        for transformed_img, transformed_batch in zip(transformed_img_list, transformed_batch_list):
+            assert_equal(transformed_img, transformed_batch[i, ...])
+
+    with get_tmp_dir() as tmp_dir:
+        scripted_fn.save(os.path.join(tmp_dir, "t_op_list_{}.pt".format(method)))
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('dt', [None, torch.float32, torch.float64])
+@pytest.mark.parametrize('size', [32, 34, [32, ], [32, 32], (32, 32), [34, 35]])
+@pytest.mark.parametrize('max_size', [None, 35, 1000])
+@pytest.mark.parametrize('interpolation', [BILINEAR, BICUBIC, NEAREST])
+def test_resize(dt, size, max_size, interpolation, device):
+
+    # TODO: Minimal check for bug-fix, improve this later
+    x = torch.rand(3, 32, 46)
+    t = T.Resize(size=38)
+    y = t(x)
+    # If size is an int, smaller edge of the image will be matched to this number.
+    # i.e, if height > width, then image will be rescaled to (size * height / width, size).
+    assert isinstance(y, torch.Tensor)
+    assert y.shape[1] == 38
+    assert y.shape[2] == int(38 * 46 / 32)
+
+    tensor, _ = _create_data(height=34, width=36, device=device)
+    batch_tensors = torch.randint(0, 256, size=(4, 3, 44, 56), dtype=torch.uint8, device=device)
+
+    if dt is not None:
+        # This is a trivial cast to float of uint8 data to test all cases
+        tensor = tensor.to(dt)
+    if max_size is not None and isinstance(size, Sequence) and len(size) != 1:
+        pass  # Not supported
+    else:
+        if isinstance(size, int):
+            script_size = [size, ]
+        else:
+            script_size = size
```

Comment:
Is there any point in parametrizing with an `int` type? This if/else will anyway convert it to a `list`.

`@pytest.mark.parametrize('size', [32, 34, [32, ], [32, 32], (32, 32), [34, 35]])`
So 32 and [32, ] are actually repeated. LMK your opinion? Maybe i am missing something?

Reply:
Good catch. I don't think this if else statement should be there. We pass all these different values because we want to test that the Resize still works if you pass different inputs. 

I would remove this if/else from the test and let the Resize be tested properly.

Reply:
Yeah, that is exactly what I had in mind. I'll update this.

Reply:
we should still make sure that we test `Resize` on `int` inputs though. But I agree we don't right now, except for this bit above that should be extracted out

Reply:
@NicolasHug  To clarify what we discuss here is removing the if/else statement from the test and leaving the parametrized input as is, so that the `Resize` can actually be tested on `int`.

Reply:
Also just to clarify the reason why this is here: script mode cannot accept int inputs, only lists. We should find a way to a) still test integer inputs, and b) still properly test the script mode

Reply:
@datumbox looks like we commented at the same time ^^

My comment above explains why we can't just simply remove the if/else: scripted mode would fail on ints

Reply:
Thanks for the flag. You are right this is the reason why on the original version Victor is forced to convert it to a list. With my proposed modification, the tests should now fail.

Reply:
Since this test really is just about the scripted mode, let's remove all ints from the parametrization.

And let's extract out this bit https://github.com/pytorch/vision/pull/4010#discussion_r648159490 where we can still test on ints. While extracting it out, hopefully we can make that bit a bit more robust (maybe by parametrizing over different ints?), but if we can't, that's fine too: there's no regression. Does that make sense @AnirudhDagar ?

Reply:
Yeah, that makes sense. I'll push changes with the updates.

Reply:
@NicolasHug Agreed with the plan. And yes we were commenting at the same time, hence the confusion. 😄